### PR TITLE
PR for Azure Usage Solution

### PR DESCRIPTION
**Pull request by Volkanco**

### CODE REVIEW

File: oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json [hunk 1]
```diff
@@ -0,0 +1,786 @@
...
+        "description": "Create new or use an existing Log Analytic Workspace"
+      }
+    },
+    "omsLogAnalyticsRegion": {
+      "type": "string",
+      "allowedValues": [
+        "westeurope",
+        "eastus",
+        "southeastasia",
+        "Australia Southeast",
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json]: please be consistent: "australiasoutheast"

File: oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json [hunk 2]
```diff
@@ -0,0 +1,786 @@
...
+    "opsInsightWorkspaceKey": "AzureUsage-OPSINSIGHTS_WS_KEY",
+    "opsInsightWorkspaceKeyType": "string",
+    "opsInsightWorkspaceKeyDescription": "Encrypted value of the user's omsWorkspaceKey",
+    "createScheduleAutomationAccountName": "AzureUsage-AzureAutomationAccount-MS-Mgmt",
+    "createScheduleAutomationAccountType": "string",
+    "createScheduleAutomationAccountDescription": "The name of the Automation Account",
+    "createScheduleResourceGroupName": "AzureUsage-AzureAutomationResourceGroup-MS-Mgmt",
+    "createScheduleResourceGroupType": "string",
+    "createScheduleResourceGroupDescription": "The name of the resource group"
+
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json]: remove whitespace

File: oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json [hunk 3]
```diff
@@ -0,0 +1,786 @@
...
+        "sku": {
+          "name": "[parameters('omsLogAnalyticsSku')]"
+        }
+      },
+      "resources": [
+        {
+          "apiVersion": "2015-11-01-preview",
+          "name": "[variables('omsSolutions').customSolution.name]",
+          "type": "views",
+          "id": "[resourceId('Microsoft.OperationalInsights/workspaces/views/', parameters('omsLogAnalyticsWorkspaceName'), variables('omsSolutions').customSolution.name)]",
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json]: "id" is not needed, hence can be removed

File: oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json [hunk 4]
```diff
@@ -0,0 +1,786 @@
...
+      },
+      "properties": {
+        "workspaceResourceId": "[resourceId('Microsoft.OperationalInsights/workspaces/', parameters('omsLogAnalyticsWorkspaceName'))]",
+        "referencedResources": [],
+        "containedResources": [
+          "[resourceId('Microsoft.OperationalInsights/workspaces/views/', parameters('omsLogAnalyticsWorkspaceName'), variables('omsSolutions').customSolution.name)]"
+        ]
+      }
+    }
+
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json]: remove whitespace =)

File: oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json (hunk 1 above)
Comment (Volkanco) [inline oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json]: done...

File: oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json (hunk 2 above)
Comment (Volkanco) [inline oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json]: Done...

File: oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json (hunk 3 above)
Comment (Volkanco) [inline oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json]: done

File: oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json (hunk 4 above)
Comment (Volkanco) [inline oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json]: done

File: oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json [hunk 5]
```diff
@@ -0,0 +1,786 @@
...
+                          "isDefault": false
+                        }
+                      ]
+                    },
+                    "NameDSVSeparator": "",
+                    "NavigationQuery": "{selected item}"
+                  }
+                }
+              }
+
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json]: another whitespace =)

File: oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json [hunk 6]
```diff
@@ -0,0 +1,786 @@
...
+                "Advanced": {
+                  "DataFlowVerification": {
+                    "Enabled": false,
+                    "Query": "*",
+                    "Message": ""
+                  }
+                }
+              }
+            }
+
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json]: plz remove whitespace

File: oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json [hunk 7]
```diff
@@ -0,0 +1,786 @@
...
+          }
+        }
+      ]
+    },
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeployonlyloganalytics.json]: you might want to "prettify" this section

File: oms-azure-resource-usage-solution/azuredeploy.parameters.json [hunk 8]
```diff
@@ -1,43 +1,42 @@
...
-    "omsAutomationAccountName": {
-      "value": "GEN-UNIQUE-30"
-    },
-    "omsAutomationRegion": {
-      "value": "eastus2"
-    },
-    "omsDataIngestionFrequency": {
-       "value": "Hourly"
-    },
-    "Currency": {
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeploy.parameters.json]: please follow camelCase pattern =)

File: oms-azure-resource-usage-solution/azuredeploy.parameters.json [hunk 9]
```diff
@@ -1,43 +1,42 @@
...
-       "value": "USD"
-    },
-    "RegionInfo": {
-      "value": "United States"
-    },
-    "Locale": {
-      "value": "en-US"
-    },
-    "OfferDurableId": {
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeploy.parameters.json]: camelCase

File: oms-azure-resource-usage-solution/azuredeploy.parameters.json [hunk 10]
```diff
@@ -1,43 +1,42 @@
...
-    "omsDataIngestionFrequency": {
-       "value": "Hourly"
-    },
-    "Currency": {
-       "value": "USD"
-    },
-    "RegionInfo": {
-      "value": "United States"
-    },
-    "Locale": {
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeploy.parameters.json]: change to "locale"

File: oms-azure-resource-usage-solution/azuredeploy.parameters.json [hunk 11]
```diff
@@ -1,43 +1,42 @@
...
+        "omsDataIngestionFrequency": {
+            "value": "Hourly"
+        },
+        "Currency": {
+            "value": "USD"
+        },
+        "RegionInfo": {
+            "value": "United States"
+        },
+        "Locale": {
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeploy.parameters.json]: change to "locale"

File: oms-azure-resource-usage-solution/azuredeployonlyautomation.json [hunk 12]
```diff
@@ -1,663 +1,662 @@
...
-      "defaultValue": "Hourly",
-      "allowedValues": [
-        "Hourly",
-        "Daily"
-      ],
-      "metadata": {
-        "description": "Specify the Azure Usage data retrieval frequency"
-      }
-    },
-    "Currency": {
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeployonlyautomation.json]: change to "currency"

File: oms-azure-resource-usage-solution/azuredeployonlyautomation.json [hunk 13]
```diff
@@ -1,663 +1,662 @@
...
-        "TRY",
-        "TWD",
-        "USD",
-        "ZAR"
-      ],
-      "metadata": {
-        "description": "Specify the Azure Billing Currency"
-      }
-    },
-    "RegionInfo": {
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeployonlyautomation.json]: change to "regionInfo"

File: oms-azure-resource-usage-solution/azuredeployonlyautomation.json [hunk 14]
```diff
@@ -1,663 +1,662 @@
...
-      "type": "string",
-      "defaultValue": "United States",
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeployonlyautomation.json]: Is there a reason for why we want to default it to United States?

File: oms-azure-resource-usage-solution/azuredeployonlyautomation.json [hunk 15]
```diff
@@ -1,663 +1,662 @@
...
-        "Vietnam",
-        "Yemen",
-        "Zambia",
-        "Zimbabwe"
-      ],
-      "metadata": {
-        "description": "Specify region for Azure BillingSee https://github.com/Azure/azure-content-nlnl/blob/master/articles/billing-countries-and-currencies.md for details."
-      }
-    },
-    "Locale": {
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeployonlyautomation.json]: change to "locale"

File: oms-azure-resource-usage-solution/azuredeployonlyautomation.json [hunk 16]
```diff
@@ -1,663 +1,662 @@
...
-        "zh-HK",
-        "zh-MO",
-        "zh-SG",
-        "zh-TW"
-      ],
-      "metadata": {
-        "description": "Specify locale for Azure Billing. See https://github.com/Azure/azure-content-nlnl/blob/master/articles/billing-countries-and-currencies.md for details."
-      }
-    },
-    "OfferDurableId": {
```
Comment (krnese) [inline oms-azure-resource-usage-solution/azuredeployonlyautomation.json]: change to "offerDurableId"

**Pull request merged**